src/commands/force/org/shape/delete.ts

```diff
@@ -0,0 +1,157 @@
+
+  public async run(): Promise<OrgShapeDeleteResult> {
+    if (!this.flags.noprompt) {
+      if (!(await this.ux.confirm(messages.getMessage('deleteCommandYesNo', [this.org.getUsername()])))) {
+        return;
+      }
+    }
+
+    this.conn = this.org.getConnection();
+
+    if (!(await isShapeEnabled(this.conn))) {
+      const err = messages.createError('noAccess', [this.org.getUsername()]);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore override readonly .name field
+      err.name = 'noAccess';
+      throw err;
+    }
+
+    const deleteRes = await this.deleteAll();
+
+    if (deleteRes.shapeIds.length === 0) {
+      this.ux.log(messages.getMessage('noShapesHumanSuccess', [this.org.getOrgId()]));
+      return;
+    }
+
+    if (deleteRes.failures.length > 0 && deleteRes.shapeIds.length > 0) {
+      this.setExitCode(68);
+
+      this.ux.styledHeader('Partial Success');
+      this.ux.log(messages.getMessage('humanSuccess', [this.org.getOrgId()]));
+      this.ux.log('');
+      this.ux.styledHeader('Failures');
+      const columns = {
+        shapeId: { header: 'Shape ID' },
+        message: { header: 'Error Message' },
+      };
+      this.ux.table(deleteRes.failures, columns);
+    } else if (deleteRes.failures.length > 0) {
+      this.setExitCode(1);
+    } else if (deleteRes.shapeIds.length > 0) {
+      this.setExitCode(0);
+      this.ux.log(messages.getMessage('humanSuccess', [this.org.getOrgId()]));
+    }
+
+    return {
+      orgId: this.org.getOrgId(),
+      shapeIds: deleteRes.shapeIds,
+      failures: deleteRes.failures,
+    };
+  }
+
+  protected setExitCode(code: number): void {
+    process.exitCode = code;
+  }
+
+  /**
+   * Delete all ShapeRepresentation records for the shapeOrg.
+   *
+   * @return List of SR IDs that were deleted
+   */
+  private async deleteAll(): Promise<DeleteAllResult> {
+    const deleteAllResult = {
+      shapeIds: [],
+      failures: [],
+    };
+
+    let shapeIds: string[] = [];
+
+    try {
+      const result = await this.conn.query<ShapeRepresentation>('SELECT Id FROM ShapeRepresentation');
+      if (result.totalSize === 0) {
+        return deleteAllResult;
+      }
+      shapeIds = result.records.map((shape) => shape.Id);
+    } catch (err) {
+      const JsForceErr = err as JsForceError;
+      if (JsForceErr.errorCode && JsForceErr.errorCode === 'INVALID_TYPE') {
+        // ShapeExportPref is not enabled, or user does not have CRUD access
+        const e = messages.createError('noAccess', [this.org.getUsername()]);
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore override readonly .name field
+        e.name = 'noAccess';
+        throw e;
+      }
+      // non-access error
+      throw JsForceErr;
+    }
+
+    await Promise.allSettled(
+      shapeIds.map(async (id) => {
+        try {
+          const delResult = await this.conn.sobject('ShapeRepresentation').delete(id);
```

Comment:
In my testing, when passing an `shapeId[]` to `delete` it doesn't return the shapeId if one fails so in case of a partial success there would be no way to match a server error to a specific shape, so I went back to `Promise.allSettled` as suggested.